### PR TITLE
Shapes

### PR DESCRIPTION
**Pull request by immabird**
added the shapes endpoint

### CODE REVIEW

**Comment by immabird** [comment]
This PR addresses issue #13

File: mbta/shapes.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	GetAllShapesSortByPriorityAscending    GetAllShapesSortByType = "priority"
+	GetAllShapesSortByPriorityDecending    GetAllShapesSortByType = "-priority"
+)
+
+// ShapeInclude is an enumerable for all the includes for shape
+type ShapeInclude string
+
+const (
+	ShapeIncludeRoute ShapeInclude = includeRoute
+	ShapeIncludeStop               = includeStop
```
Comment (mellena1) [inline mbta/shapes.go]: should be `stops` not `stop`

File: go.mod [hunk 2]
```diff
@@ -5,6 +5,7 @@ go 1.12
 require (
 	github.com/google/go-querystring v1.0.0
 	github.com/google/jsonapi v0.0.0-00010101000000-000000000000
+	github.com/twpayne/go-polyline v1.0.0 // indirect
```
Comment (mellena1) [inline go.mod]: get rid of this

File: go.sum [hunk 3]
```diff
@@ -2,5 +2,7 @@ github.com/google/go-querystring v1.0.0 h1:Xkwi/a1rcvNg1PPYe5vI8GbeBY/jrVuDX5ASu
 github.com/google/go-querystring v1.0.0/go.mod h1:odCYkC5MyYFN7vkCjXpyrEuKhc/BUO6wN/zVPAxq5ck=
 github.com/michele/go.jsonapi v0.0.0-20190314113132-fa5f73018bc4 h1:tebzQT2gdm8k8l7auUhgBtlXACalp13zsDRNDULpI7c=
 github.com/michele/go.jsonapi v0.0.0-20190314113132-fa5f73018bc4/go.mod h1:ZmncCscyUCj7nQSO8mGOYc12GMVOanri7Hj4I0DbNoE=
+github.com/twpayne/go-polyline v1.0.0 h1:EA8HPg0MNS62R5D2E8B6zyz2TMkmkXVlQaVBVgY3F5A=
```
Comment (mellena1) [inline go.sum]: and get rid of these

File: mbta/shapes.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	Sort              GetAllShapesSortByType `url:"sort,omitempty"`                 // Results can be sorted by the id or any /data/{index}/attributes key. Assumes ascending; may be prefixed with '-' for descending
+	Fields            []string               `url:"fields[shape],comma,omitempty"`  // Fields to include with the response. Multiple fields MUST be a comma-separated (U+002C COMMA, “,”) list. Note that fields can also be selected for included data types: see the V3 API Best Practices (https://www.mbta.com/developers/v3-api/best-practices) for an example.
+	Include           []ShapeInclude         `url:"include,comma,omitempty"`        // Can include choose to include route and stop.
+	FilterRoute       []string               `url:"filter[route],comma,omitempty"`  // Filter by /data/{index}/relationships/route/data/id.
+	FilterDirectionID string                 `url:"filter[direction_id],omitempty"` // Filter by direction of travel along the route.
+}
+
+// GetShapeRequestConfig holds the request info for the GetShape function
+type GetShapeRequestConfig struct {
```
Comment (mellena1) [inline mbta/shapes.go]: Move this one down to above getshape

File: mbta/shapes.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+	Fields  []string `url:"fields[shape],comma,omitempty"` // Fields to include with the response. Multiple fields MUST be a comma-separated (U+002C COMMA, “,”) list. Note that fields can also be selected for included data types: see the V3 API Best Practices for an example.
+	Include []string `url:"include,comma,omitempty"`       // Can include choose to include route and stop.
```
Comment (mellena1) [inline mbta/shapes.go]:
```suggestion
	Include []ShapeInclude `url:"include,comma,omitempty"`       // Can include choose to include route and stop.
```

File: mbta/shapes.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+// GetAllShapes gets all the shapes based on the config info
+func (s *ShapeService) GetAllShapes(config GetAllShapesRequestConfig) ([]*Shape, *http.Response, error) {
```
Comment (mellena1) [inline mbta/shapes.go]: Change all of the config parameters to pointers because we decided to change them all to pointers at one point

**Review by mellena1 — CHANGES_REQUESTED** [review]

**Review by mellena1 — APPROVED** [review]
LGTM

**Pull request merged**